osquery/events/darwin/openbsm.cpp

```diff
@@ -16,10 +18,92 @@
 
 namespace osquery {
 
+const int kQLimit = 512;
+
 DECLARE_bool(disable_audit);
+DECLARE_bool(audit_allow_process_events);
+DECLARE_bool(audit_allow_sockets);
+DECLARE_bool(audit_allow_user_events);
+DECLARE_bool(audit_allow_fim_events);
 
 REGISTER(OpenBSMEventPublisher, "event_publisher", "openbsm");
 
+Status OpenBSMEventPublisher::configureAuditPipe(FILE* au_pipe) {
```

Comment:
Since this is a member function, why do we pass the `FILE*` instead of accessing `audit_pipe_` directly?

Reply:
I was making the copy of audit_pipe_ pointer to au_pipe and setting the 
audit_pipe_ = nullptr;
doing so stuff
audit_pipe_ = au_pipe ;
return 

instead of doing
auto au_pipe =  audit_pipe_ ;
will change that to no argument function.